sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -858,3 +863,62 @@ def merge_only(self, accumulators):
 
   def extract_only(self, accumulator):
     return self.combine_fn.extract_output(accumulator)
+
+
+class Latest(object):
+  """Combiners for computing the latest element"""
+
+  class Globally(ptransform.PTransform):
+    """Compute the element with the latest timestamp from a
+    PCollection."""
+
+    @staticmethod
+    def add_timestamp(element, timestamp=core.DoFn.TimestampParam):
+      return [(element, timestamp)]
+
+    def expand(self, pcoll):
+      return (pcoll
+              | core.ParDo(self.add_timestamp)
+              .with_output_types(Tuple[T, TimestampType])
+              | core.CombineGlobally(LatestCombineFn()))
+
+  class PerKey(ptransform.PTransform):
+    """Compute elements with the latest timestamp for each key
+    from a keyed PCollection"""
+
+    @staticmethod
+    def add_timestamp(element, timestamp=core.DoFn.TimestampParam):
+      _check_instance_type(KV[K, V], element)
```

Comment:
You can drop this. The reason is, during pipeline construction these validation will happen, the best thing to do at transform level is to use typehints as much as possible to express what the transform is doing. These are limited but it is better than introducing one off checks inside transforms.

See Top transform for an example: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/combiners.py#L314


 (This will also be good for not using an internal function.)

Reply:
Thank You @aaltay! I'll go through it and make the necessary changes.